### PR TITLE
ticdc: update storage sink doc

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
update storage sink doc

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?
ref https://github.com/pingcap/tiflow/pull/8954

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 1]
```diff
@@ -160,23 +156,26 @@ CDC000005.csv
 
 ### DDL 事件
 
-当 DDL 事件引起表的版本变更时，TiCDC 将会切换到新的路径下写入数据变更记录，例如 `test.table1` 的版本从 `9999` 变更为 `10000` 时将会在 `s3://bucket/bbb/ccc/test/table1/10000/2022-01-02/CDC000001.csv` 路径中写入数据。并且，当 DDL 事件发生时，TiCDC 将生成一个 `schema.json` 文件存储表结构信息。
+#### 表级 DDL 事件
+
+当 DDL 事件引起表的版本变更时，TiCDC 将会切换到新的路径下写入数据变更记录，例如 `test.table1` 的版本变更为 `441349361156227074` 时将会在 `s3://bucket/bbb/ccc/test/table1/441349361156227074/2022-01-02/CDC000001.csv` 路径中写入数据。并且，当 DDL 事件发生时，TiCDC 将生成一个文件存储表结构信息。
 
 表结构信息将会存储到以下路径：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:164]:
```suggestion
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 2]
```diff
@@ -223,9 +223,32 @@ CDC000005.csv
     - `ColumnIsPk`：值为 `true` 时表示该列是主键的一部分。
 - `TableColumnsTotal`：`TableColumns` 数组的大小。
 
+#### 库级 DDL 事件
+
+数据库结构信息将会存储到以下路径：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:228]:
```suggestion
当上游数据库发生库级 DDL 事件时，TiCDC 将会自动生成一个文件存储数据库结构信息，文件路径如下：

```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -223,9 +223,32 @@ CDC000005.csv
...
+
+```shell
+{scheme}://{prefix}/{schema}/meta/schema_{table-version}_{hash}.json
+```
+
+一个名为 `schema_441349361156227000_3131721815.json` 的文件如下:
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:234]:
```suggestion
以 `schema_441349361156227000_3131721815.json 为例，数据库结构信息文件的内容如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 4]
```diff
@@ -234,7 +257,7 @@ TiDB 中整数类型可被定义为 `IT[(M)] [UNSIGNED]`，其中：
 - `IT` 为整数类型，包括 `TINYINT`、`SMALLINT`、`MEDIUMINT`、`INT`、`BIGINT` 和 `BIT`。
 - `M` 为该类型的显示宽度。
 
-`schema.json` 文件中对整数类型定义如下：
+schema  文件中对整数类型定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:260]:
```suggestion
schema 文件中对整数类型定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 5]
```diff
@@ -252,7 +275,7 @@ TiDB 中的小数类型可被定义为 `DT[(M,D)][UNSIGNED]`，其中：
 - `M` 为该类型数据的精度，即整数位加上小数位的总长度。
 - `D` 为小数位的长度。
 
-`schema.json` 文件中对小数类型的定义如下：
+schema  文件中对小数类型的定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:278]:
```suggestion
schema 文件中对小数类型的定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 6]
```diff
@@ -269,7 +292,7 @@ TiDB 中的日期类型可被定义为 `DT`，其中：
 
 - `DT` 为日期类型，包括 `DATE` 和 `YEAR`。
 
-`schema.json` 文件中对日期类型的定义如下：
+schema  文件中对日期类型的定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:295]:
```suggestion
schema 文件中对日期类型的定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 7]
```diff
@@ -283,7 +306,7 @@ TiDB 中的时间类型可被定义为 `TT[(M)]`，其中：
 - `TT` 为时间类型，包括 `TIME`、`DATETIME` 和 `TIMESTAMP`。
 - `M` 为秒的精度，取值范围为 0~6。
 
-`schema.json` 文件中对时间类型的定义如下：
+schema  文件中对时间类型的定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:309]:
```suggestion
schema 文件中对时间类型的定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 8]
```diff
@@ -300,7 +323,7 @@ TiDB 中的字符串类型可被定义为 `ST[(M)]`，其中：
 - `ST` 为字符串类型，包括 `CHAR`、`VARCHAR`、`TEXT`、`BINARY`、`BLOB`、`JSON` 等。
 - `M` 表示字符串的最大长度。
 
-`schema.json` 文件中对字符串类型的定义如下：
+schema  文件中对字符串类型的定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:326]:
```suggestion
schema 文件中对字符串类型的定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 9]
```diff
@@ -312,7 +335,7 @@ TiDB 中的字符串类型可被定义为 `ST[(M)]`，其中：
 
 ### Enum/Set 类型
 
-`schema.json` 文件中对 Enum/Set 类型的定义如下：
+schema  文件中对 Enum/Set 类型的定义如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:338]:
```suggestion
schema 文件中对 Enum/Set 类型的定义如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 10]
```diff
@@ -223,9 +223,32 @@ CDC000005.csv
...
+  "Type": 1,
+  "TableColumns": null,
+  "TableColumnsTotal": 0
+}
+```
+
 ## 数据类型
 
-本章节主要介绍 `schema.json` 文件中使用的各种数据类型。数据类型定义为 `T(M[, D])`，详见[数据类型概述](/data-type-overview.md#数据类型概述)。
+本章节主要介绍 schema 文件中使用的各种数据类型。数据类型定义为 `T(M[, D])`，详见[数据类型概述](/data-type-overview.md#数据类型概述)。
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:251]:
```suggestion
本章节主要介绍 `schema_{table-version}_{hash}.json` 文件（以下简称为 schema 文件）中使用的各种数据类型。数据类型定义为 `T(M[, D])`，详见[数据类型概述](/data-type-overview.md#数据类型概述)。
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 11]
```diff
@@ -118,18 +118,14 @@ NFS 配置样例如下：
 
 > **注意：**
 >
-> 表的版本可能在以下三种情况下发生变化：
->
-> - 发生 DDL 操作后，表的版本为该 DDL 在上游 TiDB 执行结束的 TSO。但是，表版本的变化并不意味着表结构的变化。例如，在表中的某一列添加注释，不会导致 `schema.json` 文件内容发生变化。
-> - 进程重启，表的版本为进程重启时 Changefeed 的 checkpoint TSO。在有很多表的情况下，重启时需要遍历所有目录并找到上一次重启时每张表写入的位置，这样的操作耗时较长。因此，TiCDC 选择在一个以 checkpoint TSO 为版本的新目录下写入数据，而不是在旧版本的目录下继续写入数据。
-> - 发生表调度后，表的版本为该表调度到当前节点时 Changefeed 的 checkpoint TSO。
+> 表的版本仅在发生 DDL 操作后才改变：表的版本为该 DDL 在上游 TiDB 执行结束的 TSO。但是，表版本的变化并不意味着表结构的变化。例如，在表中的某一列添加注释，不会导致 schema 文件内容发生变化。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply ticdc/ticdc-sink-to-cloud-storage.md:121]: 问题：上游表库级别的 DDL 操作会改成表的版本？

**Comment by Lloyd-Pottiger** [comment]
should also update `ticdc/ticdc-storage-consumer-dev-guide.md`?

File: ticdc/ticdc-sink-to-cloud-storage.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (CharlesCheung96) [reply ticdc/ticdc-sink-to-cloud-storage.md:121]: 库级 DDL 不会影响表版本

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 12]
```diff
@@ -160,23 +156,27 @@ CDC000005.csv
 
 ### DDL 事件
 
-当 DDL 事件引起表的版本变更时，TiCDC 将会切换到新的路径下写入数据变更记录，例如 `test.table1` 的版本从 `9999` 变更为 `10000` 时将会在 `s3://bucket/bbb/ccc/test/table1/10000/2022-01-02/CDC000001.csv` 路径中写入数据。并且，当 DDL 事件发生时，TiCDC 将生成一个 `schema.json` 文件存储表结构信息。
+#### 表级 DDL 事件
+
+当上游表的 DDL 事件引起表的版本变更时，TiCDC 将会自动进行以下操作：
 
-表结构信息将会存储到以下路径：
+- 切换到新的路径下写入数据变更记录。例如，当 `test.table1` 的版本变更为 `441349361156227074` 时，TiCDC 将会在 `s3://bucket/bbb/ccc/test/table1/441349361156227074/2022-01-02/CDC000001.csv` 路径中写入数据。
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:163]:
```suggestion
- 切换到新的路径下写入数据变更记录。例如，当 `test.table1` 的版本变更为 `441349361156227074` 时，TiCDC 将会在 `s3://bucket/bbb/ccc/test/table1/441349361156227074/2022-01-02/` 路径下写入数据。
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 13]
```diff
@@ -160,23 +156,27 @@ CDC000005.csv
...
+- 生成一个文件存储表结构信息，文件路径如下：
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:164]:
```suggestion
- 生成一个 schema 文件存储表结构信息，文件路径如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 14]
```diff
@@ -223,9 +224,33 @@ CDC000005.csv
     - `ColumnIsPk`：值为 `true` 时表示该列是主键的一部分。
 - `TableColumnsTotal`：`TableColumns` 数组的大小。
 
+#### 库级 DDL 事件
+
+当上游数据库发生库级 DDL 事件时，TiCDC 将会自动生成一个文件存储数据库结构信息，文件路径如下：
+
+
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:231]:
```suggestion
```

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 15]
```diff
@@ -223,9 +224,32 @@ CDC000005.csv
     - `ColumnIsPk`：值为 `true` 时表示该列是主键的一部分。
 - `TableColumnsTotal`：`TableColumns` 数组的大小。
 
+#### 库级 DDL 事件
+
+当上游数据库发生库级 DDL 事件时，TiCDC 将会自动生成一个文件存储数据库结构信息，文件路径如下：
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:229]:
```suggestion
当上游数据库发生库级 DDL 事件时，TiCDC 将会自动生成一个 schema 文件存储数据库结构信息，文件路径如下：
```

**Review by qiancai — APPROVED** [review]
/lgtm

**Comment by CharlesCheung96** [comment]
> should also update `ticdc/ticdc-storage-consumer-dev-guide.md`?

Thanks, we'll deal with that in another pr.

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 16]
```diff
@@ -213,6 +213,7 @@ CDC000005.csv
 - `Version`：Storage sink 协议版本号。
 - `TableVersion`：表的版本号。
 - `Query`：DDL 语句。
+- `Type`: DDL 类型。
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:216]:
```suggestion
- `Type`：DDL 类型。
```
用户可以通过 Type 数字 `5` 知道具体对应的是什么 DDL 类型吗，有没有相关的 Type 数字->类型映射的文档链接呢

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 17]
```diff
@@ -223,9 +224,32 @@ CDC000005.csv
...
 
+#### 库级 DDL 事件
+
+当上游数据库发生库级 DDL 事件时，TiCDC 将会自动生成一个 schema 文件存储数据库结构信息，文件路径如下：
+
+```shell
+{scheme}://{prefix}/{schema}/meta/schema_{table-version}_{hash}.json
+```
+
+以 `schema_441349361156227000_3131721815.json 为例，数据库结构信息文件的内容如下：
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-cloud-storage.md:235]:
```suggestion
以 `schema_441349361156227000_3131721815.json` 为例，数据库结构信息文件的内容如下：
```

File: ticdc/ticdc-sink-to-cloud-storage.md (hunk 16 above)
Comment (CharlesCheung96) [reply ticdc/ticdc-sink-to-cloud-storage.md]: 这个与 TiDB 中保持一致，这里只是作为参考

**Review by Oreoxmt — COMMENTED** [review]
/lgtm

**Pull request merged**